File: frontend/src/services/reviewService.ts
import { apiClient } from '../lib/apiClient'

export interface ReviewResponse {
  id: number
  productId: number
  userId: number
  rating: number
  commentText: string
  createdAt: string
  updatedAt: string
}

export interface CreateReviewRequest {
  productId: number
  rating: number
  commentText: string
}

export const reviewService = {
  getByProductId(productId: number): Promise<ReviewResponse[]> {
    return apiClient.get<ReviewResponse[]>(`/api/reviews/products/${productId}`)
  },

  create(request: CreateReviewRequest): Promise<ReviewResponse> {
    return apiClient.post<ReviewResponse>('/api/reviews', request)
  },

  delete(id: number): Promise<void> {
    return apiClient.delete<void>(`/api/reviews/${id}`)
  },
}

File: backend/review-service/src/main/java/com/yabozkurt/n11bootcamp/ecommerce/review/application/service/ReviewService.java
package com.yabozkurt.n11bootcamp.ecommerce.review.application.service;

import com.yabozkurt.n11bootcamp.ecommerce.review.presentation.dto.request.CreateReviewRequest;
import com.yabozkurt.n11bootcamp.ecommerce.review.presentation.dto.request.UpdateReviewRequest;
import com.yabozkurt.n11bootcamp.ecommerce.review.presentation.dto.response.ReviewResponse;

import java.util.List;

public interface ReviewService {

    ReviewResponse create(Long userId, CreateReviewRequest request);

    ReviewResponse getById(Long id);

    List<ReviewResponse> getByProductId(Long productId);

    List<ReviewResponse> getByUserId(Long userId);

    ReviewResponse update(Long userId, Long id, UpdateReviewRequest request);

    void delete(Long userId, Long id);

    void deleteByAdmin(Long id);
}
